R/temporal_torus_translation.R

```diff
@@ -62,45 +87,43 @@ cyclic_shift <- function(df, time.var="year", species.var="species",  abundance.
 #' @import stats
 #' @export
-confint.cyclic_shift <- function(df, time.var="year", species.var="species",  abundance.var="abundance", FUN, bootnumber,
-                                 li=0.025, ui=0.975, replicate.var=NA, average.replicates=T){
-  if(!is.numeric(df[[abundance.var]])) { stop("Abundance variable is not numeric") }
+confint.cyclic_shift <- function(object, parm = "out", level = 0.95, ...){
+
+  li <- (1 - level)/2
+  ui <- 1 - li
+
+  out <- object[[parm]]
+
+#   if (is.na(replicate.var)) {
 
-  if(is.na(replicate.var)){
-    check_single_onerep(df, time.var, species.var)
-    out <- replicate(bootnumber, FUN(shuffle_community(transpose_community(df, time.var,  species.var, abundance.var))))
     lowerCI <- quantile(out, li)
-    upperCI <-quantile(out, ui)
+    upperCI <- quantile(out, ui)
     nullmean <- mean(out)
     output <- cbind(lowerCI, upperCI, nullmean)
     row.names(output) <- NULL
 
-  } else {
-    df[replicate.var] <- if(is.factor(df[[replicate.var]]) == TRUE){factor(df[[replicate.var]])} else {df[replicate.var]}
-    check_single(df, time.var, species.var, replicate.var)
-    df <- df[order(df[[replicate.var]]),]
-    X <- split(df, df[replicate.var])
-    lout <- lapply(X, cyclic_shift, time.var, species.var, abundance.var, FUN, bootnumber)
-
-    if(average.replicates  ==  TRUE){
-      out <- do.call("rbind", lout)
-      out <- colMeans(out)
-      lowerCI <- quantile(out, li)
-      upperCI <-quantile(out, ui)
-      nullmean <- mean(out)
-      output <- as.data.frame(cbind(lowerCI, upperCI, nullmean))
-      row.names(output) <- NULL
-    }
-
-    else {
-      lowerCI <- do.call("rbind", lapply(lout, quantile, li))
-      upperCI <- do.call("rbind", lapply(lout, quantile, ui))
-      nullmean <- do.call("rbind", lapply(lout, mean))
-      reps <- unique(df[replicate.var])
-      output <- cbind(reps, lowerCI, upperCI, nullmean)
-      names(output)[2:3]=c("lowerCI", "upperCI")
-    }
-  }
+  # } else {
+  #
+  #   ## simple workaround, a model for how this function will need to work
+  #   lout <- lapply(lout, `[[`, i = "out")
+  #
+  #   if (average.replicates  ==  TRUE) {
+  #     out <- do.call("rbind", lout)
+  #     out <- colMeans(out)
+  #     lowerCI <- quantile(out, li)
+  #     upperCI <- quantile(out, ui)
+  #     nullmean <- mean(out)
+  #     output <- as.data.frame(cbind(lowerCI, upperCI, nullmean))
+  #     row.names(output) <- NULL
+  #   } else {
+  #     lowerCI <- do.call("rbind", lapply(lout, quantile, li))
+  #     upperCI <- do.call("rbind", lapply(lout, quantile, ui))
+  #     nullmean <- do.call("rbind", lapply(lout, mean))
+  #     reps <- unique(df[replicate.var])
+  #     output <- cbind(reps, lowerCI, upperCI, nullmean)
+  #     names(output)[2:3] <- c("lowerCI", "upperCI")
+  #   }
+  # }
```

Comment:
Yes, since the confint does it

On Fri, Mar 18, 2016 at 1:00 AM, Andrew MacDonald <notifications@github.com>
wrote:

> In R/temporal_torus_translation.R
> <https://github.com/laurenmh/codyn/pull/67#discussion_r56619100>:
>
> > +  #     out <- do.call("rbind", lout)
> > +  #     out <- colMeans(out)
> > +  #     lowerCI <- quantile(out, li)
> > +  #     upperCI <- quantile(out, ui)
> > +  #     nullmean <- mean(out)
> > +  #     output <- as.data.frame(cbind(lowerCI, upperCI, nullmean))
> > +  #     row.names(output) <- NULL
> > +  #   } else {
> > +  #     lowerCI <- do.call("rbind", lapply(lout, quantile, li))
> > +  #     upperCI <- do.call("rbind", lapply(lout, quantile, ui))
> > +  #     nullmean <- do.call("rbind", lapply(lout, mean))
> > +  #     reps <- unique(df[replicate.var])
> > +  #     output <- cbind(reps, lowerCI, upperCI, nullmean)
> > +  #     names(output)[2:3] <- c("lowerCI", "upperCI")
> > +  #   }
> > +  # }
>
> all this should probably be deleted
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/laurenmh/codyn/pull/67/files/f3e38639882cb3e07b489b62de4bd6214853b215#r56619100>
>